homeassistant/components/lektrico/config_flow.py

```diff
@@ -0,0 +1,142 @@
+"""Config flow for Lektrico Charging Station."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from lektricowifi import Device, DeviceConnectionError
+import voluptuous as vol
+
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import (
+    ATTR_HW_VERSION,
+    ATTR_SERIAL_NUMBER,
+    CONF_HOST,
+    CONF_NAME,
+    CONF_TYPE,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+
+class LektricoFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a Lektrico config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize flow."""
+        self._host: str
+        self._name: str
+        self._serial_number: str
+        self._board_revision: str
+        self._device_type: str
+
+    async def async_step_user(
+        self, user_input: dict[str, str] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._async_show_setup_form()
+
+        self._host = user_input[CONF_HOST]
+
+        # obtain serial number
+        try:
+            await self._get_lektrico_device_settings_and_treat_unique_id()
+        except DeviceConnectionError:
+            return self._async_show_setup_form(errors={CONF_HOST: "cannot_connect"})
+
+        return self._async_create_entry()
+
+    @callback
+    def _async_show_setup_form(
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, str] | None = None,
+    ) -> ConfigFlowResult:
+        """Show the setup form to the user."""
+        if user_input is None:
+            user_input = {}
+
+        schema = self.add_suggested_values_to_schema(STEP_USER_DATA_SCHEMA, user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=schema,
+            errors=errors or {},
+        )
+
+    @callback
+    def _async_create_entry(self) -> ConfigFlowResult:
+        return self.async_create_entry(
+            title=self._name,
+            data={
+                CONF_HOST: self._host,
+                CONF_NAME: self._name,
```

Comment:
Why do we store the name in the config entry data?

Reply:
I take the name from config entry data and use it when I __init__ DataUpdateCoordinator:
`super().__init__(
            hass,
            LOGGER,
            name=friendly_name,
            update_interval=SCAN_INTERVAL,
        )`
and in sensor DeviceInfo.
The name has the format:   type_serial   => I can remove CONF_NAME from entry data and create the name in those two places where I need the name.

Reply:
I think its better to create the name in place yes

Reply:
I removed CONF_NAME from entry data and created the name in those two places, thanks.